kafka/adminapi.go

```diff
@@ -1048,6 +1055,54 @@ type ListOffsetsResult struct {
 	ResultInfos map[TopicPartition]ListOffsetsResultInfo
 }
 
+// ElectionType represents the type of election to be performed
+type ElectionType int
+
+const (
+	// ElectionTypePreferred - Preferred election type
+	ElectionTypePreferred ElectionType = C.RD_KAFKA_ELECTION_TYPE_PREFERRED
+	// ElectionTypeUnclean - Unclean election type
+	ElectionTypeUnclean ElectionType = C.RD_KAFKA_ELECTION_TYPE_UNCLEAN
+)
+
+// ElectionTypeFromString translates an election type name to
+// an ElectionType value.
+func ElectionTypeFromString(electionTypeString string) (ElectionType, error) {
+	switch strings.ToUpper(electionTypeString) {
+	case "PREFERRED":
+		return ElectionTypePreferred, nil
+	case "UNCLEAN":
+		return ElectionTypeUnclean, nil
+	default:
+		return ElectionTypePreferred, NewError(ErrInvalidArg, "Unknown election type", false)
+	}
+}
+
+// ElectLeadersRequest holds parameters for the type of election to be performed and
+// the topic partitions for which election has to be performed
+type ElectLeadersRequest struct {
+	// Election type to be performed
+	electionType ElectionType
+	// TopicPartitions for which election has to be performed
+	partitions []TopicPartition
+}
+
+// NewElectLeadersRequest creates a new ElectLeadersRequest with the given election type
+// and topic partitions
+func NewElectLeadersRequest(electionType ElectionType, partitions []TopicPartition) ElectLeadersRequest {
+	return ElectLeadersRequest{
+		electionType: electionType,
+		partitions:   partitions,
+	}
+}
+
+// ElectLeadersResult holds the result of the election performed
+type ElectLeadersResult struct {
+	// TopicPartitions for which election has been performed and the per-partition error, if any
+	// that occurred while running the election for the specific TopicPartition.
+	topicPartitions []TopicPartition
```

Comment:
This must be made public
```suggestion
	TopicPartitions []TopicPartition
```